### PR TITLE
feat(paddle): Add frontend index_sample to search.py

### LINKED ISSUE #23165: index_sample

### PR DESCRIPTION
**Pull request by ILB-96**
# PR Description
- New Paddle frontend index_sample function was added to search.py.
- Test for the new function in test_search.py

## Related Issue

# 23165

Close #23165

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by a0m0rajab** [comment]
Hi @Hazannovich thank you for the PR, I just changed the title and removed a space at the beginning of it, which caused the semantic checks to fail. 

Thank you for contributing to ivy though!

File: ivy/functional/frontends/paddle/search.py [hunk 1]
```diff
@@ -33,6 +33,15 @@ def argsort(x, /, *, axis=-1, descending=False, name=None):
     return ivy.argsort(x, axis=axis, descending=descending)
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("int32", "int64", "float32", "float64")},
+    "paddle",
+)
+@to_ivy_arrays_and_back
+def index_sample(x, index):
```
Comment (YushaArif99) [inline ivy/functional/frontends/paddle/search.py:41]: the signature of `index_select` does not match the original function's signature [here](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/index_select_cn.html#paddle.index_select)

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_search.py [hunk 2]
```diff
@@ -148,6 +148,37 @@ def test_paddle_argsort(
     )
 
 
+@handle_frontend_test(
+    fn_tree="paddle.index_sample",
+    array_indices_axis=helpers.array_indices_axis(
+        array_dtypes=helpers.get_dtypes("valid"),
+        indices_dtypes=helpers.get_dtypes("integer"),
+        min_num_dims=2,
+        max_num_dims=2,
```
Comment (YushaArif99) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_search.py:157]: the array and axis generation strategy also seems quite restrictive to me. The input `x` in [paddle.index_select](https://github.com/PaddlePaddle/Paddle/blob/release/2.5/python/paddle/tensor/search.py#L312) can be any n-dim tensor so there's no need to limit it to 2D. 
You'll also need to add the generation strategy for the `axis` parameter which can be in the range of `[-dim, dim-1]` where `dim` is the dimension of the input `x`

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @ILB-96 

Apologies for the delay in review. Had been a little swamped with work recently and so haven't really had the time to review any PRs 😅 

Could you kindly address the abovementioned points. Feel free to re-request a review once these have been resolved and all relevant tests still pass. Thanks 🙂

**Review by YushaArif99 — APPROVED** [review]
Apologies for the previous comments. It turns out I was reviewing the wrong function here 😂. I thought this PR was implementing `paddle.index_select` which is slightly different from `paddle.index_sample` (which is what the PR is **actually** implementing). 

With this out of the way, the changes look good to me now 👍🏼

**Pull request merged**